fingerprint.c

```diff
@@ -286,14 +324,10 @@ static int fingerprint_enroll(struct fingerprint_device *dev,
 
     fpc_verify_auth_challenge(sdev->fpc, (void*) hat, sizeof(hw_auth_token_t));
 
-    pthread_mutex_lock(&sdev->lock);
-    sdev->worker.thread_running  = true;
-    pthread_mutex_unlock(&sdev->lock);
-
-    if(pthread_create(&sdev->worker.thread, NULL, enroll_thread_loop, (void*)sdev)) {
-        ALOGE("%s : Error creating thread\n", __func__);
-        sdev->worker.thread_running  = false;
-        return FINGERPRINT_ERROR;
+    if (getState(sdev) == STATE_ENROLL) {
```

Comment:
It feels like this check should be in setState because then it can be actually mutex-protected.
You will presumably need to make setState return and error value of some kind though.